### ISSUE TITLE
Registrar #1 Display name and status incorrect

### ISSUE THREAD

**Issue by chevdor** [polkascan/polkascan-explorer-gui#14]
Location: https://polkascan.io/pre/kusama/account/registrars
Registrar #1 shows as:
![image](https://user-images.githubusercontent.com/738724/79439868-a0c13680-7fd5-11ea-8c16-d3420cdeb92f.png)

In the PolkadotJS  UI it shows properly:
![image](https://user-images.githubusercontent.com/738724/79439923-b33b7000-7fd5-11ea-91cc-524e420ab9b7.png)

Beside, I would be interested in knowing where this `Tester` is coming from.

**Comment by chevdor** [comment]
Any news on this issue?

It is rather alarming if the UI of Polkascan does not show 100% the state of the chain and it raises the question whether some other discrepancies exist between displayed values and on-chain state. 

Please note that Polkascan does see the registrar verification:
https://polkascan.io/pre/kusama/account/Fom9M5W6Kck1hNAiE2mDcZ67auUCiNTzLBUdQy4QnxHSxdn#identity

![image](https://user-images.githubusercontent.com/738724/79737181-3a9d2200-82fb-11ea-9a63-8ab22eede75a.png)

So there is a bug somewhere.

I don't recall ever using `Tester` anywhere so I really wonder were this value comes from.

**Comment by arjanz** [comment]
I ran some tests and locally it shows the correct identity, so not sure why it shows like that on Polkascan, we are still investigating.. Obviously we have our 'mini state-machine' that could get out of sync when events or calls change due to runtime upgrades, but this doesn't seem to be the issue in this case. We will keep you updated if this is a bug or some faulty data was unintentionally synced

**Comment by chevdor** [comment]
Any news on this issue ?

**Comment by chevdor** [comment]
I see you managed to remove the "tester" label on Registrar #1

![image](https://user-images.githubusercontent.com/738724/82213742-f1012080-9914-11ea-9815-352c9b4b8e54.png)

However, display name and verification status are still missing. Did you identify the issue?

**Comment by chevdor** [comment]
Any news ?

**Comment by chevdor** [comment]
The issue or a similar one also shows up on CC1.
Some definitely verified accounts are showing a wrong status despite the judgement being visible to Polkascan.

Some examples below:
![image](https://user-images.githubusercontent.com/738724/87965984-e56cda00-cabc-11ea-88c7-180ab0f63da8.png)

![image](https://user-images.githubusercontent.com/738724/87966014-f0276f00-cabc-11ea-9de3-ed96a9b03395.png)


![image](https://user-images.githubusercontent.com/738724/87966059-ff0e2180-cabc-11ea-827e-a3f5c3f38985.png)

![image](https://user-images.githubusercontent.com/738724/87966083-06352f80-cabd-11ea-8d97-3ca8c481b539.png)


![image](https://user-images.githubusercontent.com/738724/87966108-0fbe9780-cabd-11ea-8a59-617ed3fdfb66.png)

![image](https://user-images.githubusercontent.com/738724/87966141-1ea54a00-cabd-11ea-9cd0-7e422bc54d3f.png)


Hint, the issue seems to be showing up for judgements set thru a proxy with `ProxyType` = `IdentityJudgement` .

**Comment by chevdor** [comment]
As far as Kusama is concerned, there was a change, now no status is visible at all for Registrar #1:
![image](https://user-images.githubusercontent.com/738724/87966419-8b204900-cabd-11ea-8a5e-e7af42f4152f.png)

**Comment by emielvanderhoek** [comment]
We are aware of those problems. This requires substantial refactoring and will be scheduled sometime in August.

**Comment by chevdor** [comment]
Can you explain what the issue is and why it affects some accounts and not some others ?

**Comment by arjanz** [comment]
It has to do with the fact that we use the data from the referenced extrinsic of the `IdentitySet` and `JudgementGiven` events to determine what the identity data is or what judgement is given. But with the introduction of batched calls and sudo calls, this mechanism gets confused because the actual data is wrapped in a `Vec` or positioned elsewhere in the call. You can see this as an 'internal call', sort of same idea with the internal transaction of Ethereum. 

We are aware of this shortcoming, but refactoring to extract all the calls from the extrinsics in a uniform way will take some time unfortunately.. 

So basically, currently it should work if the call of the extrinsic is `Identity.set_indentity` or `Indentity.provide_judgement`, but not yet if the call in the extrinsic is a `Utility.batch` or `Sudo.sudo`.

**Comment by chevdor** [comment]
Thanks for the explanation, that totally makes sense, and you can probably add `proxy.*` calls to the list.
